plugins/inputs/intel_powerstat/README.md

```diff
@@ -0,0 +1,199 @@
+# Intel PowerStat Input Plugin
+
+Telemetry frameworks allow users to monitor critical platform level metrics. 
+Key source of platform telemetry is power domain that is beneficial for MANO/Monitoring&Analytics systems 
+to take preventive/corrective actions based on platform busyness, CPU temperature, actual CPU utilization 
+and power statistics. Main use cases are power saving and workload migration.
+
+Intel PowerStat plugin supports Intel based platforms and assumes presence of Linux based OS.
+
+### Configuration:
+```toml
+# Intel PowerStat plugin enables monitoring of platform metrics (power, TDP) and per-CPU metrics like temperature, power and utilization.
+[[inputs.intel_powerstat]]
+  ## All global metrics are always collected by Intel PowerStat plugin.
+  ## User can choose which per-CPU metrics are monitored by the plugin in cpu_metrics array.
+  ## Empty array means no per-CPU specific metrics will be collected by the plugin - in this case only platform level
+  ## telemetry will be exposed by Intel PowerStat plugin.
+  ## Supported options:
+  ## "cpu_frequency", "cpu_c1_state_residency", "cpu_c6_state_residency", "cpu_busy_cycles", "cpu_temperature", "cpu_busy_frequency"
+  # cpu_metrics = []
+```
+### Example: Configuration with no per-CPU telemetry
+This configuration allows getting global metrics (processor package specific), no per-CPU metrics are collected:
+```toml
+[[inputs.intel_powerstat]]
+  cpu_metrics = []
+```
+
+### Example: Configuration for CPU Temperature and Frequency only
+This configuration allows getting global metrics plus subset of per-CPU metrics (CPU Temperature and Current Frequency):
+```toml
+[[inputs.intel_powerstat]]
+  cpu_metrics = ["cpu_frequency", "cpu_temperature"]
+```
+
+### Example: Configuration with all available metrics
+This configuration allows getting global metrics and all per-CPU metrics:
+```toml
+[[inputs.intel_powerstat]]
+  cpu_metrics = ["cpu_frequency", "cpu_c1_state_residency", "cpu_c6_state_residency", "cpu_busy_cycles", "cpu_temperature", "cpu_busy_frequency"]
+```
+
+### SW Dependencies:
+Plugin is based on Linux Kernel modules that expose specific metrics over `sysfs` or `devfs`.
+The following dependencies are expected by plugin:
+- _intel-rapl_ module which exposes Intel Runtime Power Limiting metrics over `sysfs` (`/sys/devices/virtual/powercap/intel-rapl`)
+Minimum kernel version required is 3.13 (to make sure RAPL driver is included in current distro),
+- _msr_ kernel module that provides access to processor model specific registers from user space, 
+- _cpufreq_ kernel module - which exposes per-CPU Frequency over `sysfs`. Since kernel 3.4 the necessary modules are 
+loaded automatically and the recommended on demand governor is enabled by default.
+
+Please make sure that kernel modules are loaded and running. You might have to manually enable them by using `modprobe`.
+Plugin requires root access to read MSR files. 
```

Comment:
This mean the whole telegraf needs to run as root right? Ouch! Maybe make it a bold hint!

Reply:
Yep, it should be bolded.

Reply:
On my machine everything in /sys/devices/virtual/powercap/intel-rapl is world readable.  I think this means the plugin would work without root access.  I'd be surprised if other linux distributions had different permissions.

A quick search of similar tools also seems to imply root access isn't needed. e.g. http://web.eece.maine.edu/~vweaver/projects/rapl/

I'd strongly prefer not to encourage users to run telegraf as root, especially if it's not necessary.  What exactly needs root access?  Does the plugin write to files in sysfs to configure RAPL?

Reply:
@reimda 
You're right - to read `/sys/devices/virtual/powercap/intel-rapl` you don't need root access (probably on every distro).

But here is written: `Plugin requires root access to read MSR files.` which are located here: `/dev/cpu/%s/msr` (`%s` for logical cpu core). And this is the character device file to which you need the root access (no workarounds for that :()

We will try change README here to show what can be read without root and for which metrics root is required.

(and no, plugin does not write to any file)

Reply:
@srebhan corrected.
@reimda I found out that on some environments (OS/conf) meaningful files under `/sys/devices/virtual/powercap/intel-rapl` (text files, not directories) need root access and on some they don't. Section restructured.